### DISCUSSION TITLE
some operations fail in remote repo, outputting only ^G

### DISCUSSION THREAD

**Discussion by ahihi** [magit/magit#5267] [Support]
Magit v4.1.3-5-g873da4a6, Transient 0.8.1, Git 2.47.0, Emacs 30.0.92, darwin

i am accessing a remote repo hosted on a macOS 10.14 machine. magit is set up with `emacs -Q` as follows:

```elisp
(defvar bootstrap-version)
(let ((bootstrap-file
       (expand-file-name
        "straight/repos/straight.el/bootstrap.el"
        (or (bound-and-true-p straight-base-dir)
            user-emacs-directory)))
      (bootstrap-version 7))
  (unless (file-exists-p bootstrap-file)
    (with-current-buffer
        (url-retrieve-synchronously
         "https://raw.githubusercontent.com/radian-software/straight.el/develop/install.el"
         'silent 'inhibit-cookies)
      (goto-char (point-max))
      (eval-print-last-sexp)))
  (load bootstrap-file nil 'nomessage))

(straight-use-package 'magit)
```

the system git on the remote is outdated, so i point to a separately installed 2.47.0:

```elisp
(setq magit-remote-git-executable "/run/current-system/sw/bin/git")
```

status works normally, as does adding files (`s`). however, other operations such as commit (`c c`) and pull (`F p`) appear to start but never produce any result.

in `magit-process` i see the following strange output:

```
run git … pull origin refs/heads/master
^G^G^G^G^G^G^G^G^G^G^G^G^G^G^G^G^G^G^G^G^G^G^G^G^G^G^G^G^G^G^G^G^G^G^G^G^G^G^G^G^G^G^G^G^G^G^G^G
  0 git … add -- test.txt
run git … commit --
^G^G^G^G^G^G^G^G^G^G^G^G^G^G^G^G^G^G^G^G^G^G^G^G^G^G^G^G
```

running the `pull` and `add` commands in the shell (with `magit-git-global-arguments` included) works normally, though.

**Comment by tarsius** [comment]
> i am accessing a remote repo

Using tramp?

Anyway, despite the `emacs -Q`, I suspect there is something wrong with your setup, as opposed to a bug in magit. I cannot think of a possible cause right now.

Please also note that for the rest of the year, I am reducing the time I spend on support, to have more time with family and doing end-of-year wrap-ups. I recommend you investigate the issue yourself, you are in the best position to do it, since you can reproduce it. If you treat it as a holiday challenge, you may even learn something and have some fun along the way.

**Reply by ahihi** [reply]
> Using tramp?

yes.

> Anyway, despite the emacs -Q, I suspect there is something wrong with your setup, as opposed to a bug in magit. I cannot think of a possible cause right now.

possibly. it worked before, the things that changed recently are updating Emacs from 29 to 30 and also updating all my packages.

> Please also note that for the rest of the year, I am reducing the time I spend on support, to have more time with family and doing end-of-year wrap-ups. I recommend you investigate the issue yourself, you are in the best position to do it, since you can reproduce it. If you treat it as a holiday challenge, you may even learn something and have some fun along the way.

totally understandable! i do intend to look into this myself, but am also short on time right now :)

**Reply by ahihi** [reply]
i keep running into this on various Mac systems, so i set out to eliminate the possibility of anything wrong with my setup with the following test case:

1. install a clean macOS (i am using 10.15 because it was easy to set up in a VM, but have seen this problem on 12, 13 and 14 as well)
2. install Xcode command line tools (for git)
3. install Emacs 30.1.90
4. using the snippet from the original post, install latest Magit (v4.3.8-8-g9654f5e0)
5. create/clone any git repo on the fresh system and stage a change
6. run `magit-commit-create` locally
  - the commit message buffer opens as expected. cancel the commit
7. enable the system SSH daemon
8. open the same git repo via Tramp, using a `/ssh:` path to the system's own network address
9. run `magit-commit-create`, now via Tramp
  - the commit message buffer does not open, and the `magit-process` buffer again shows a bunch of `^G`

i spent some time trying to debug this with edebug, `tramp-verbose`, `magit-toggle-git-debug` and `magit-remote-git-executable` pointed at a git wrapper script that logs some debug info. the main thing i learned from this is that the `^G`s are not coming from git, as they appear whether or not i call it in the wrapper script.

i then tested the same setup, except with Emacs 29.4. this works fine. so, should i consider this a Tramp bug?

**Comment by svenssonaxel** [comment]
Possibly related: https://github.com/magit/magit/issues/5292

@ahihi Using the debugging methods described in this issue might help you determine whether you are experiencing a bug in magit or not.
